### PR TITLE
retry mechanism for ca remote copy

### PR DESCRIPTION
**Pull request by jschicktanz**
**What this PR does / why we need it**:
add retry to component-archive remote copy cmd

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
- add retry to component-archive remote copy cmd
```

### CODE REVIEW

File: pkg/commands/componentarchive/remote/copy.go [hunk 1]
```diff
@@ -404,6 +414,28 @@ func (c *Copier) Copy(ctx context.Context, name, version string) error {
 	return nil
 }
 
+func (c *Copier) Copy(ctx context.Context, name, version string) error {
+	log := logr.FromContextOrDiscard(ctx).WithValues("component", name, "version", version)
+
+	for retries := uint64(0); retries <= c.MaxRetries; retries++ {
+		err := c.copy(ctx, name, version)
+		if err == nil {
```
[the remark this answers is not in the document]
Comment (jschicktanz) [reply pkg/commands/componentarchive/remote/copy.go:422]: already doing that in line 431 :)

File: docs/reference/component-cli_component-archive_remote_copy.md [hunk 2]
```diff
@@ -20,13 +20,15 @@ component-cli component-archive remote copy COMPONENT_NAME VERSION --from SOURCE
 
 ```
       --allow-plain-http                    allows the fallback to http if the oci registry does not support https
+      --backoff-factor duration             a backoff factor to apply between retry attempts. if backoff-factor is 1s, then the timeout will be [1s, 2s, 4s, …] (default 1s)
```
[the remark this answers is not in the document]
Comment (jschicktanz) [reply docs/reference/component-cli_component-archive_remote_copy.md]: done

File: pkg/commands/componentarchive/remote/copy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (enrico-kaack-comp) [reply pkg/commands/componentarchive/remote/copy.go:422]: you're right, disregard that

**Review by enrico-kaack-comp — APPROVED** [review]
lgtm

**Pull request merged**